test/transport.test.js

```diff
@@ -0,0 +1,158 @@
+    target: pinoSyslog,
+    level: 'info',
+    options: sysLogOptions
+  })
+  const log = pino(transport)
+  t.pass('built pino')
+  await once(transport, 'ready')
+  t.pass('transport ready ' + destination)
+
+  log.info(JSON.parse(messages.leadingDay))
+  log.debug(JSON.parse(messages.helloWorld)) // it is skipped
+  log.info(JSON.parse(messages.trailingDay))
+
+  await timeout(1000)
+
+  const data = fs.readFileSync(destination, 'utf8').trim().split('\n')
+  t.ok(data[0].startsWith('<134>Feb  3 01:20:00 MacBook-Pro-3 none[94473]: '), true, 'first line leadingDay')
+  t.ok(data[1].startsWith('<134>Feb 10 01:20:00 MacBook-Pro-3 none[94473]: '), true, 'first line trailingDay')
+})
+
+test('syslog pino transport test rfc5424', async t => {
+  const destination = join(os.tmpdir(), 'pino-transport-test.log')
+
+  const fd = fs.openSync(destination, 'w+')
+  const sysLogOptions = {
+    destination: fd,
+    ...getConfigPath('5424', 'newline.json')
+  }
+
+  const transport = pino.transport({
+    target: pinoSyslog,
+    level: 'info',
+    options: sysLogOptions
+  })
+  const log = pino(transport)
+  t.pass('built pino')
+  await once(transport, 'ready')
+  t.pass('transport ready ' + destination)
+
+  log.info(JSON.parse(messages.leadingDay))
+  log.debug(JSON.parse(messages.helloWorld)) // it is skipped
+  log.info(JSON.parse(messages.trailingDay))
+
+  await timeout(1000)
+
+  const data = fs.readFileSync(destination, 'utf8').trim().split('\n')
+  t.ok(data[0].startsWith('<134>1 2018-02-03T01:20:00Z MacBook-Pro-3 - 94473 - - '), 'first line leadingDay')
+  t.ok(data[1].startsWith('<134>1 2018-02-10T01:20:00Z MacBook-Pro-3 - 94473 - - '), 'first line trailingDay')
+})
+
+test('syslog pino transport test stdout', async t => {
+  const result = spawnSync('node', ['--no-warnings', join(__dirname, 'fixtures', 'log-stdout.js'), '1'], {
+    cwd: process.cwd()
+  })
+  t.equal(result.stdout.toString().trim(), '<134>1 2016-04-01T16:44:58Z MacBook-Pro-3 - 94473 - - hello world')
+  t.equal(result.status, 0)
+})
+
+test('syslog pino transport test stderr', async t => {
+  const result = spawnSync('node', ['--no-warnings', join(__dirname, 'fixtures', 'log-stdout.js'), '2'], {
+    cwd: process.cwd()
+  })
+  t.equal(result.stderr.toString().trim(), '<134>1 2016-04-01T16:44:58Z MacBook-Pro-3 - 94473 - - hello world')
+  t.equal(result.status, 0)
+})
+
+test('pino pipeline', t => {
+  t.plan(4)
+  const destination = join(os.tmpdir(), 'pino-transport-test.log')
+
+  const expected = [
+    '<134>1 2018-02-03T01:20:00Z MacBook-Pro-3 - 94473 - - ',
+    '<134>1 2018-02-10T01:20:00Z MacBook-Pro-3 - 94473 - - '
+  ]
+
+  createTcpListener(msg => {
+    msg.split('\n')
+      .filter(line => line) // skip empty lines
+      .forEach(line => {
+        t.ok(line.startsWith(expected.shift()))
+      })
+  })
+    .then((serverSocket) => {
+      t.teardown(() => {
+        serverSocket.close()
+        serverSocket.unref()
+        transport.end()
```

Comment:
I had to add this line to make it works on node12.

I thought it was due the `duplexyfy` stream, that accepts these two parameters:

```
      autoDestroy: true,
      destroy: true
```

But it was not.
Nether the underlining `socket.unref()` was the issue.

I would like to understand why the transport doesn't close. 
I still don not figured it out.

Reply:
What was the problem? why the test did not pass?

Reply:
`tap` triggers a timeout error - locally within an infinity timeout, the process does not end without any errors